File: algorithms/js/palindromeNumber/palindromeNumber.js
// Source : https://leetcode-cn.com/problems/palindrome-number/
// Author : Tian Zhuo
// Date   : 2018-11-17
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x < 0) return false;
    let rev = 0;
    let orig = x;
    while(x !== 0){
        rev = rev * 10 + x % 10;
        x = Math.floor(x / 10);
    }
    return orig === rev;
};

/**
 * @param {number} x
 * @return {boolean}
 */
let isPalindrome1 = function(x) {
    if (x < 0 || (x % 10 === 0 && x !== 0)) return false;
    let revert = 0;
    while (x > revert) { // this condition reduce the loop to half! optimization!
        revert = (revert * 10) + x % 10;
        x = Math.floor(x / 10);
    }
    return x === revert || x === Math.floor(revert / 10);
};

File: src/main/java/palindromeNumber/PalindromeNumber.java
/*
 * @Source: https://leetcode-cn.com/problems/palindrome-number/
 * @Author: Tian Zhuo
 * @Date: 2018-12-14 21:36:09
 */
package palindromeNumber;

public class PalindromeNumber {
    public boolean isPalindrome(int x) {
        if(x < 0 || (x != 0 && x % 10 == 0)) return false;
        int revert = 0;
        while(x > revert) { //只用转换一半即可
            revert = revert * 10 + x % 10;
            x = x / 10;
        }
        return x == revert || x == revert / 10; //偶数位 与 奇数位
    }
}